### ISSUE TITLE
How to create a context menue on a TreeItem?

### ISSUE THREAD

**Issue by DavidBal** [vscode-elements/elements#205]
Hi,

i would like to generate a custome context menue on Item of Tree.

![image](https://github.com/user-attachments/assets/b15bfc0c-d37c-4a39-8611-f84e4312d4a7)

Is there a way to archive this?

Thanks in advance!

Kind regards,
David

**Comment by neko-para** [comment]
Generally, the context item itself is kind of seperate window, then the major problem seems to be the way to listen to the right click of the items. However, #152 should definitely solve this problem, thus waiting for it would be a better choice, rather than request a feature for the deprecated tree widget 🤔

**Comment by bendera** [comment]
> Hi,
> 
> i would like to generate a custome context menue on Item of Tree.

Is this what you are looking for?
https://code.visualstudio.com/api/extension-guides/webview#context-menus

**Comment by DavidBal** [comment]
> > Hi,
> > i would like to generate a custome context menue on Item of Tree.
> 
> Is this what you are looking for? https://code.visualstudio.com/api/extension-guides/webview#context-menus

Yes, but i see no way to place the "data-vscode-context" on the element via the provided [api](https://vscode-elements.github.io/components/tree/api/#custom-types).

> Generally, the context item itself is kind of seperate window, then the major problem seems to be the way to listen to the right click of the items. However, #152 should definitely solve this problem, thus waiting for it would be a better choice, rather than request a feature for the deprecated tree widget 🤔

That looks intressting, i can wait.

-----

Thanks.

**Comment by bendera** [comment]
I'm not sure if it works but you can try something like this:
```javascript
document.querySelector('vscode-tree').shadowRoot.querySelector('ul:nth-child(2) li').dataset.vscodeContext = '{"webviewSection": "editor", "preventDefaultContextMenuItems": true}'
```

**Comment by bendera** [comment]
@DavidBal v2 preview has been published: https://github.com/vscode-elements/elements/releases/tag/v2.0.0-pre.2
